src/main/java/org/opensearch/knn/plugin/script/KNNScoringSpace.java

```diff
@@ -44,28 +46,20 @@ public interface KNNScoringSpace {
     ScoreScript getScoreScript(Map<String, Object> params, String field, SearchLookup lookup, LeafReaderContext ctx, IndexSearcher searcher)
         throws IOException;
 
-    class L2 implements KNNScoringSpace {
-
+    /**
+     * Base class to represent linear space
```

Comment:
What do you mean by linear space?

Reply:
Any suggestion? 
L2, inner product, and cosine similarity are vector space(linear space) and hamming is its one space. I choose linear because I we are calling hamming as one of vector space.

Reply:
Oh I see. Then yes I think thats okay